src/Microsoft.TemplateEngine.Edge/FileSystem/InMemoryFileSystem.cs

```diff
@@ -0,0 +1,917 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+
+namespace Microsoft.TemplateEngine.Edge.FileSystem
+{
+    internal class InMemoryFileSystem : IPhysicalFileSystem, IFileLastWriteTimeSource
```

Comment:
I think this implementations are stable and useful for other hosts, why do we want this internal/not in Utils?

Reply:
was in doubt for this, but yes probably better to keep them public. I would prefer to keep them in Edge probably, what do you think?